web/packages/teleport/src/Assist/context/AssistContext.tsx

```diff
@@ -448,15 +471,41 @@ export function AssistContextProvider(props: PropsWithChildren<unknown>) {
     const url = cfg.getAssistExecuteCommandUrl(
       getHostName(),
       clusterId,
-      getAccessToken(),
       execParams
     );
 
-    const proto = new Protobuf();
     executeCommandWebSocket.current = new WebSocket(url);
-    executeCommandWebSocket.current.binaryType = 'arraybuffer';
 
+    executeCommandWebSocket.current.onopen = () => {
+      executeCommandWebSocket.current.send(
+        JSON.stringify({ token: getAccessToken() })
+      );
+    };
```

Comment:
Seems to me that we have a a similar sort of thing in several places, wherein we have a `websocket.onopen` which needs to send the json-ified access token, and then parse the expected response in a certain way `onmessage` before flipping an `authenticated` flag. This makes me think that we might want to provide a shared `AuthenticatingWebSocket` that wraps a native `WebSocket` object and keeps all this logic in one place, and then swap these various individual `WebSocket`s out for that.

We'll have to get the API right so that you can basically use it like an ordinary `WebSocket`, but with our little authentication protocol automatically sprinkled on top; It might take a bit of work, but it will be a good way to keep the implementation of authenticating websockets consistent throughout the codebase going forward. WDYT?

Reply:
This would be a great solution, especially as it seems we need some logic to wait for an authenticated connection before attempting to send any messages

Reply:
I'm not particularly confident when it comes to good javascript interface design, but I've pushed a version with something like this and a version working when connecting to a node, lmk what suggestions ye might have.